### DISCUSSION TITLE
Multiple Device Logins

### DISCUSSION THREAD

**Discussion by Lockie85** [babybuddy/babybuddy#802] [Support]
Hey,

It appears as if (might not actually work this way) you're unable to be logged in on more than one device. As whenever I want to view babybuddy from another device, phone, tablet, PC etc. I have to login again. In addition this can even apply to the same device if/when moving that device from the local network to access babybuddy via the internet.

If there is not already a way around this, could support be provided so that a user can be logged in on multiple devices at once?

Many Thanks

**Comment by cdubz** [comment]
@Lockie85 are you able to reproduce this issue with the demo? I am able to log in there from multiple devices without having to log back in. Could this be a setting in your browser preventing cookies?

**Comment by Lockie85** [comment]
> @Lockie85 are you able to reproduce this issue with the demo? I am able to log in there from multiple devices without having to log back in. Could this be a setting in your browser preventing cookies?

I can't appear to reproduce it via the demo. It's worth noting I'm using the app via Home Assistant. Not sure if that would make a difference?

**Comment by cdubz** [comment]
Hmmm it might. Though I still suspect something browser related.

After you log in can you check for browser cookie store for a `sessionid` cookie on your Baby Buddy instance domain? And is it still there after you close and open the browser?

**Reply by Lockie85** [reply]
It's worth noting that I use the Home Assistant mobile companion/application. I'm not sure if this utilises chromium or is entirely Home Assistants own application. But the log out issue tends to happen within this application when moving from my home wifi to a cellular connection.

**Reply by MrApplejuice** [reply]
Hey Lockie85!

Maybe I can provide some background here on what is _maybe_ going on. I am not sure what you are _exactly_ doing or how you run babybuddy on home-assistant, but I would assume that you are using the lowest path of resistance and have downloaded the home assistant app and use the "side bar link" in HomeAssistant to access BabyBuddy?

Like this:

![image](https://github.com/babybuddy/babybuddy/assets/542105/e437a32b-74be-41d6-8862-fa3605b4520a)

If so, you are likely using the ingress server which adds another type of authentication on top of baby buddy, namely the home-assistant authentication itself! That will be coupled to bay buddy using an "ingress-server session cookie" if you are interested in the technical issue. I was trying to get around that issue for the android app, but it only works "half way"...

## Solutions

The trick now is that you probably need to duplicate the authentication with the ingress-server and not babybuddy. Alternatively, you can bypass the ingress-server and access babybuddy directly.

### Access baby buddy directly

I assume that this is the easiest to configure! You "just expose the ports" as explained on the addon configuration page:

![image](https://github.com/babybuddy/babybuddy/assets/542105/3b972fa3-84ce-4b91-954e-f4426f9b0d91)

This will bypass the entire ingress-stuff and you will only use baby buddy authentication and baby buddy features at that point which probably is the easiest way!

### (Could work, but untested!) Create multiple home assistant users

Creating multiple home assistant users should enable you to create different ingress sessions for the different users. This means that you should not be logged out at all and you can use one user per device.

However, most of this is resting on conjecture of what your actual issues are... as mentioned above, I think "Access baby buddy directly" is probably the go-to hassle-free solution for you!
